### ISSUE TITLE
Make an RGB number input for editor.

### ISSUE THREAD

**Issue by Sheshank-s** [griffpatch/Scratch3-Dev-Tools#8]
I'm making this now on my branch. I'll send a merge request soon.

**Comment by Sheshank-s** [comment]
Sorry I started this, and then i had schol work. I'll get it done by the end of the day for sure! 
Also @griffpatch can you confirm this idea?

**Comment by griffpatch** [comment]
No hurry don't fret :) - Could you outline what your idea is in a little more detail?

**Comment by Sheshank-s** [comment]
Yeah!
![image](https://user-images.githubusercontent.com/11379161/65286168-6cdf6a80-daf3-11e9-8a98-18af35147de2.png)

When you input the hex code it updates the other things. I made the html already im just working on the js now

**Comment by griffpatch** [comment]
Cool!

**Comment by Sheshank-s** [comment]
Weird, since the popup is a react container, adding HTML to it breaks the popup (sliders don't work).

**Comment by Sheshank-s** [comment]
Never mind that ^ I found a work around

**Comment by Sheshank-s** [comment]
Hmm... I'm not able to change the color. I tried simulating the slider moving and then a click but the click won't happen at the place i want, it only goes to the beginning of the slider.

**Comment by Sheshank-s** [comment]
Do you have any ideas on how i can get it to work?

**Comment by ghost** [comment]
I don't know, but try to use the function `setColor(color)`
and look at https://github.com/LLK/scratch-paint/blob/develop/src/containers/color-picker.jsx

**Comment by Sheshank-s** [comment]
How would I be able to access that function from the extension though?

**Comment by griffpatch** [comment]
Interfacing with the existing framework can be really difficult, if not
impossible... I am having great trouble finding creatine ways of doing
things!

**Comment by Sheshank-s** [comment]
@griffpatch Yeah... One option is we can set the left value of the slider to the correct places, and then the user would need to move one of the sliders a little bit to the left so that it updates... but that's a really crappy way of doing it.

**Comment by Sheshank-s** [comment]
@griffpatch @RaspberryPiFan Doing something like this
`document.getElementsByClassName("slider_container_o2aIb")[0].click()`
Resets the slider to 0. If there is a way to click at a certain position on the slider, that might work.

**Comment by Sheshank-s** [comment]
Another idea, we can generate an image in the extension, so that the user can click the extension, put in hex value, copy image, paste in scratch editor, then use color picker.

That's really time consuming for user though..

**Comment by griffpatch** [comment]
> @griffpatch @RaspberryPiFan Doing something like this
> `document.getElementsByClassName("slider_container_o2aIb")[0].click()`
> Resets the slider to 0. If there is a way to click at a certain position on the slider, that might work.

Don't reference the class name ending in _o2aIb because this could change with each build of scratch. Instead you need to locate the element using a selector like this:

let find = document.querySelector('[class*=slider_container_]')

**Comment by Sheshank-s** [comment]
@griffpatch Okay cool. Should I just make it so that it says underneath the hex code what the color brightness and saturation should be?

**Comment by Sheshank-s** [comment]
![image](https://user-images.githubusercontent.com/11379161/65389774-519c7700-dd0e-11e9-88b4-187182d0c84d.png)
And then the user has to move the sliders, that should work!

**Comment by Explosion-Scratch** [comment]
What if we use the existing framework for the color picker to simulate using it?
